daemon/control.c

```diff
@@ -620,6 +626,50 @@ control_handle_message_unpriv(const ControllerToDaemon *msg, int fd)
 	}
 }
 
+/**
+ * Starts a container with pre-specified keys or user supplied keys
+ */
+static int
+control_handle_container_start(control_t *control, container_t *container,
+			       ContainerStartParams *start_params, int fd)
+{
+	TRACE("Starting container");
+	int res = -1;
+
+	if (start_params && start_params->has_setup) {
+		INFO("Setting Setup mode for Container!");
+		container_set_setup_mode(container, start_params->setup);
+	}
+
+	// Check if pin should be interactively requested via pin pad reader
+	if (container_get_usb_pin_entry(container)) {
+		TRACE("Container start with pin entry chosen. Starting");
+		res = input_register_container_start_cb(control, container, start_params);
```

Comment:
keep the protobuf types encapsulated in this file. the input system does not need the start params
anyway. Since it reads the key from the input device itself. Thus, just remove the start_params paramter here
and in the typedef inside the input subsystem.

Reply:
Done